client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx

```diff
@@ -0,0 +1,397 @@
+		const { expired } = domain;
+
+		if ( expired && ! this.isDomainInGracePeriod() ) {
+			return null;
+		}
+
+		return (
+			<DomainManagementNavigationItem
+				path={ domainManagementTransfer( selectedSite.slug, domain.name ) }
+				materialIcon="sync_alt"
+				text={ translate( 'Transfer domain' ) }
+				description={ translate( 'Transfer lock: off' ) }
+			/>
+		);
+	}
+
+	getDomainConnectMapping() {
+		const { selectedSite, translate, domain } = this.props;
+
+		const { supportsDomainConnect, hasWpcomNameservers, pointsToWpcom } = domain;
+
+		if ( ! supportsDomainConnect || hasWpcomNameservers || pointsToWpcom ) {
+			return;
+		}
+
+		const path = domainManagementDomainConnectMapping( selectedSite.slug, domain.name );
+
+		return (
+			<DomainManagementNavigationItem
+				path={ path }
+				materialIcon="cached"
+				text={ translate( 'Connect your domain' ) }
+			/>
+		);
+	}
+
+	handleChangeSiteAddressClick = () => {
+		const { domain } = this.props;
+		const domainType = getDomainTypeText( domain );
+
+		this.props.recordGoogleEvent(
+			'Domain Management',
+			`Clicked "Change Site Address" navigation link on a ${ domainType } in Edit`,
+			'Domain Name',
+			domain.name
+		);
+
+		this.props.recordTracksEvent( 'calypso_domain_management_change_navigation_click', {
+			action: 'change_site_address',
+			section: domain.type,
+		} );
+	};
+
+	handlePickCustomDomainClick = () => {
+		const { domain } = this.props;
+		const domainType = getDomainTypeText( domain );
+
+		this.props.recordGoogleEvent(
+			'Domain Management',
+			`Clicked "Pick a custom domain" navigation link on a ${ domainType } in Edit`,
+			'Domain Name',
+			domain.name
+		);
+
+		this.props.recordTracksEvent( 'calypso_domain_management_wpcom_domain_add_domain', {
+			action: 'change_site_address',
+			section: domain.type,
+		} );
+	};
+
+	getPickCustomDomain() {
+		const { selectedSite, translate } = this.props;
+
+		return (
+			<DomainManagementNavigationItem
+				path={ domainAddNew( selectedSite.slug ) }
+				onClick={ this.handlePickCustomDomainClick }
+				materialIcon="search"
+				text={ translate( 'Pick a custom domain' ) }
+			/>
+		);
+	}
+
+	getSiteAddressChange() {
+		const { domain, selectedSite, translate } = this.props;
+		const { isWpcomStagingDomain } = domain;
+
+		if ( isWpcomStagingDomain ) {
+			return;
+		}
+
+		return (
+			<DomainManagementNavigationItem
+				path={ domainManagementChangeSiteAddress( selectedSite.slug, domain.name ) }
+				onClick={ this.handleChangeSiteAddressClick }
+				materialIcon="create"
+				text={ translate( 'Change site address' ) }
+			/>
+		);
+	}
+
+	getSimilarDomains() {
+		const { selectedSite, translate } = this.props;
+
+		return (
+			<DomainManagementNavigationItem
+				path={ domainAddNew( selectedSite.slug ) }
+				onClick={ this.handlePickCustomDomainClick }
+				materialIcon="search"
+				text={ translate( 'Find similar domains' ) }
+				description={ translate( 'Placeholder text' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **26** times:
`translate( 'Set the placeholder text' )` *ES Score: 7*